pkg/hooks/api.go

```diff
@@ -0,0 +1,24 @@
+package hooks
+
+import (
+	"encoding/json"
+
+	k8sv1 "k8s.io/api/core/v1"
+)
+
+const HookSidecarListAnnotationName = "hookSidecars"
+
+type HookSidecarList []HookSidecar
+
+type HookSidecar struct {
```

Comment:
Maybe we need to use the container struct from the kubernetes deployment?

Reply:
for now, users shouldn't have the ability to define the sidecar container other than saying indicating the image they want to use. virt-controller then constructs the actual container spec in a precise way that the users do not have control over. 

this is up for debate though if there's we can come up with a good reason to expose that level of control to users.